src/LanguageServer/Protocol.TestUtilities/LanguageServer/AbstractLanguageServerProtocolTests.cs

```diff
@@ -531,44 +532,71 @@ private static LSP.DidCloseTextDocumentParams CreateDidCloseTextDocumentParams(U
                }
            };
 
-        internal sealed class TestLspServer : IAsyncDisposable
+        /// <summary>
+        /// Implementation of <see cref="AbstractTestLspServer{TWorkspace, TDocument, TProject, TSolution}"/>
+        /// using the <see cref="LspTestWorkspace"/> workspace.
+        /// </summary>
+        internal sealed class TestLspServer : AbstractTestLspServer<LspTestWorkspace, TestHostDocument, TestHostProject, TestHostSolution>
         {
-            public readonly EditorTestWorkspace TestWorkspace;
-            private readonly Dictionary<string, IList<LSP.Location>> _locations;
+            public TestLspServer(LspTestWorkspace testWorkspace, Dictionary<string, IList<LSP.Location>> locations, InitializationOptions initializationOptions, AbstractLspLogger logger)
+                : base(testWorkspace, locations, initializationOptions, logger)
+            {
+            }
+
+            public static async Task<TestLspServer> CreateAsync(LspTestWorkspace testWorkspace, InitializationOptions initializationOptions, AbstractLspLogger logger)
+            {
+                var locations = await GetAnnotatedLocationsAsync(testWorkspace, testWorkspace.CurrentSolution);
+                var server = new TestLspServer(testWorkspace, locations, initializationOptions, logger);
+                await server.InitializeAsync();
+                return server;
+            }
+        }
+
+        internal abstract class AbstractTestLspServer<TWorkspace, TDocument, TProject, TSolution> : IAsyncDisposable
```

Comment:
refactored TestLspServer a bit so it can be passed any implementation of `TestWorkspace<TDocument, TProject, TSolution>`.  mostly this is just the LspTestWorkspace, but as above there are a few cases where we still use EditorTestWorkspace.